sqlmesh/core/context_diff.py

```diff
@@ -197,93 +219,93 @@ def has_snapshot_changes(self) -> bool:
         return bool(self.added or self.removed_snapshots or self.modified_snapshots)
 
     @property
-    def added_materialized_models(self) -> t.Set[str]:
+    def added_materialized_snapshot_ids(self) -> t.Set[SnapshotId]:
         """Returns the set of added internal models."""
         return {
-            name
-            for name in self.added
-            if self.snapshots[name].model_kind_name
-            and self.snapshots[name].model_kind_name.is_materialized  # type: ignore
+            s_id
+            for s_id in self.added
+            if self.snapshots[s_id].model_kind_name
+            and self.snapshots[s_id].model_kind_name.is_materialized  # type: ignore
         }
 
     @property
-    def promotable_models(self) -> t.Set[str]:
+    def promotable_snapshot_ids(self) -> t.Set[SnapshotId]:
         """The set of model names that have to be promoted in the target environment."""
         return {
-            *self.previously_promoted_model_names,
+            *self.previously_promoted_snapshot_ids,
             *self.added,
             *self.modified_snapshots,
         } - set(self.removed_snapshots)
 
     @property
-    def unpromoted_models(self) -> t.Set[str]:
-        """The set of model names that have not yet been promoted in the target environment."""
-        return set(self.snapshots) - self.previously_promoted_model_names
+    def unpromoted_models(self) -> t.Set[SnapshotId]:
+        """The set of snapshot IDs that have not yet been promoted in the target environment."""
+        return set(self.snapshots) - self.previously_promoted_snapshot_ids
 
-    def directly_modified(self, name: str) -> bool:
-        """Returns whether or not a node was directly modified in this context.
+    def directly_modified(self, s_id: SnapshotId) -> bool:
+        """Returns whether or not a snapshot was directly modified in this context.
 
         Args:
-            name: The node name to check.
+            s_id: The snapshot id to check.
```

Comment:
That's what I'm talking about. Which snapshot ID is this? The source or the target?